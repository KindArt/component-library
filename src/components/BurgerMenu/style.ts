import styled, { css } from 'styled-components';
import { Colours } from '../../constants/styles';

interface IWrapper {
  isOpen?: boolean;
}

export const BurgerMenuWrapper = styled.div<IWrapper>`
  height: 20px;
  width: 20px;
  position: relative;
  color: ${Colours.TEXT_SECONDARY};

  ${({ isOpen }) => isOpen && css``}
`;

interface ILongLineSVG {
  isOpen: boolean;
}

export const TopLine = styled.svg<ILongLineSVG>`
  position: absolute;
  fill: currentColor;
  transform: rotate(0);
  transition: 0.2s top 0.4s, 0.1s left 0.2s, 0.2s transform, 0.2s opacity 0.2s;

  left: 0;
  top: 2px;

  ${({ isOpen }) =>
    isOpen &&
    css`
      transition: 0.2s opacity, 0.2s top, 0.2s left 0.2s, 0.2s transform 0.4s;

      right: auto;
      left: 3px;
      top: 9px;

      transform: rotate(-45deg);
    `}
`;

export const MiddleLine = styled.svg<ILongLineSVG>`
  position: absolute;
  fill: currentColor;
  transform: rotate(0);
  transition: 0.2s top 0.4s, 0.1s left 0.2s, 0.2s transform, 0.2s opacity 0.2s;

  top: 9px;
  left: 0;

  ${({ isOpen }) =>
    isOpen &&
    css`
      transition: 0.2s opacity, 0.2s top, 0.2s left 0.2s, 0.2s transform 0.4s;

      opacity: 0;
    `}
`;

export const BottomLine = styled.svg<ILongLineSVG>`
  position: absolute;
  fill: currentColor;
  transform: rotate(0);
  transition: 0.2s top 0.4s, 0.1s left 0.2s, 0.2s transform, 0.2s opacity 0.2s;

  left: 0;
  top: 16px;

  ${({ isOpen }) =>
    isOpen &&
    css`
      transition: 0.2s opacity, 0.2s top, 0.2s left 0.2s, 0.2s transform 0.4s;
      right: auto;
      left: 3px;
      top: 9px;

      transform: rotate(45deg);
    `}
`;
