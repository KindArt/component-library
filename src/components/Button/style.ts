import { ReactNode } from 'react';
import styled, { css } from 'styled-components';
import { ButtonSize, IButtonProps } from './';
import { Colours } from '../../constants/styles';

interface IIconWrapper {
  icon?: ReactNode;
  iconRight?: boolean;
  iconOnly: boolean;
}

export const IconWrapper = styled.span<IIconWrapper>`
  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

interface IButtonWrapper extends IButtonProps {
  iconOnly: boolean;
}

export const ButtonWrapper = styled.button<IButtonWrapper>`
  display: inline-flex;
  align-items: center;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  height: 38px;
  padding: 0 18px;
  transition: all 0.1s ease-in-out 0s;
  font-size: 0.875rem;
  font-weight: 400;
  position: relative;

  i {
    opacity: .8;
  }

  ${({ icon }) =>
    icon &&
    css`
      padding-left: 40px;
      ${IconWrapper} {
        position: absolute;
        top: 50%;
        left: 12px;
        transform: translateY(-50%);
        font-size: 18px;
      }
    `}

  ${({ icon, iconRight }) =>
    icon &&
    iconRight &&
    css`
      padding-right: 40px;
      padding-left: 18px;
      ${IconWrapper} {
        left: auto;
        right: 12px;
      }
    `}

  ${({ iconOnly }) =>
    iconOnly &&
    css`
      padding: 0;
      justify-content: center;
      flex-basis: 38px;
      width: 38px;
      flex-shrink: 0;
      ${IconWrapper} {
        left: 50%;
        transform: translateX(-50%) translateY(-50%);
      }
    `}

  ${({ fullWidth, icon }) => {
    if (fullWidth && icon) {
      return css`
        width: 100%;
        padding-left: 40px;
        padding-right: 40px;
      `;
    } else if (fullWidth) {
      return css`
        width: 100%;
      `;
    }
  }}

  &>* {
    transition: .1s all;
  }

  ${({ size }) =>
    size === ButtonSize.Small &&
    css`
      height: 28px;
      padding: 0 15px;
      font-size: 0.75rem;
    `}

  ${({ size }) =>
    size === ButtonSize.Large &&
    css`
      height: 48px;
      padding: 0 40px;
      font-size: 1rem;
    `}

  &:not(:disabled) {
    cursor: pointer;
  }

  &:focus {
    outline: none;
  }

  &:active {
    border: 1px solid #eee;
    box-shadow: none;
    transform: translateY(0px);
  }

  /* Basic Appearence */
  background: #FFFFFF;
  background-image: linear-gradient(0deg, #F6F7F9 0%, #FFFFFF 100%);
  border: 1px solid $borderColorDarker;
  box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.05),
  inset 0 2px 0 0 rgba(255, 255, 255, 0.05);
  color: $textMain;
  border-radius: 4px;

  &:hover {
    background-image: linear-gradient(0deg, #F6F7F9 0%, #FFFFFF 100%);
    border: 1px solid #CACEDB;
    box-shadow: 0 1px 2px 0 rgba(22, 29, 37, 0.10), inset 0 2px 0 0 rgba(255, 255, 255, 0.05);
  }

  &:focus {
    background-image: linear-gradient(0deg, #F6F7F9 0%, #FFFFFF 100%);
    border: 1px solid #BDC2D1;
    box-shadow: 0 0 1px 0 rgba(22, 29, 37, 0.10), inset 0 2px 0 0 rgba(255, 255, 255, 0.05);
  }

  ${({ noBorder }) =>
    noBorder &&
    css`
      border: none !important;
      background: white;
      box-shadow: none;
      &:hover {
        box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.13);
      }
      &:focus {
        box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.1);
      }
    `}
  
  /* Colors */
  ${({ primary, noBorder }) =>
    primary &&
    css`
      color: white;
      background: ${({ theme }) => theme.buttonWrapper.primary.background};
      background-image: linear-gradient(-1deg, #1665d8 2%, #1f6fe5 98%);
      background-image: linear-gradient(0deg, #1665d8 0%, #1f6fe5 100%);
      border: 1px solid #1461d2;
      box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.1), inset 0 2px 0 0 rgba(255, 255, 255, 0.06);
      &:hover {
        background-image: linear-gradient(0deg, #115ecf 0%, #1767db 100%);
        border: 1px solid #0e57c2;
        ${noBorder
          ? `box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.18);`
          : `box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.18), inset 0 2px 0 0 rgba(255, 255, 255, 0.06);`}
      }
      &:focus {
        background-image: linear-gradient(0deg, #0e55bd 0%, #0f5bca 100%);
        border: 1px solid #0b4aa5;
        ${noBorder
          ? `box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.10);`
          : `box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.10), inset 0 2px 0 0 rgba(255, 255, 255, 0.06);`}
      }
    `}

  ${({ warning, noBorder }) =>
    warning &&
    css`
      color: white;
      background: ${({ theme }) => theme.buttonWrapper.warning.background};
      background-image: linear-gradient(-180deg, #e73c1e 0%, #cf3014 100%);
      border: 1px solid #ba0b15;
      box-shadow: 0 1px 1px 0 rgba(43, 26, 24, 0.1), inset 0 2px 0 0 rgba(255, 255, 255, 0.08);
      &:hover {
        background-image: linear-gradient(-180deg, #df3517 0%, #c42b10 100%);
        border: 1px solid #b00b14;
        ${noBorder
          ? `box-shadow: 0 1px 1px 0 rgba(43, 26, 24, 0.18);`
          : `box-shadow: 0 1px 1px 0 rgba(43, 26, 24, 0.18), inset 0 2px 0 0 rgba(255, 255, 255, 0.08);`}
      }
      &:focus {
        background-image: linear-gradient(-179deg, #c22b10 0%, #b8280e 100%);
        border: 1px solid #a00a13;
        border-radius: 4px;
        ${noBorder
          ? `box-shadow: 0 1px 1px 0 rgba(43, 26, 24, 0.10);`
          : `box-shadow: 0 1px 1px 0 rgba(43, 26, 24, 0.10), inset 0 2px 0 0 rgba(255, 255, 255, 0.08);`}
      }
    `}

  ${({ success, noBorder }) =>
    success &&
    css`
      color: white;
      background: ${({ theme }) => theme.buttonWrapper.success.background};
      background-image: linear-gradient(0deg, #34aa44 0%, #38b249 100%);
      border: 1px solid #2d9c3c;
      box-shadow: 0 1px 1px 0 rgba(19, 31, 21, 0.1), inset 0 2px 0 0 rgba(255, 255, 255, 0.06);
      &:hover {
        background-image: linear-gradient(0deg, #2ca13c 2%, #2fae40 100%);
        border: 1px solid #259835;
        ${noBorder
          ? `box-shadow: 0 1px 1px 0 rgba(19, 31, 21, 0.18);`
          : `box-shadow: 0 1px 1px 0 rgba(19, 31, 21, 0.18), inset 0 2px 0 0 rgba(255, 255, 255, 0.06);`}
      }
      &:focus {
        background-image: linear-gradient(0deg, #259835 0%, #27a337 100%);
        border: 1px solid #1a8628;
        ${noBorder
          ? `box-shadow: 0 1px 1px 0 rgba(19, 31, 21, 0.10);`
          : `box-shadow: 0 1px 1px 0 rgba(19, 31, 21, 0.10), inset 0 2px 0 0 rgba(255, 255, 255, 0.06);`}
      }
    `}

  ${({ transparent }) =>
    transparent &&
    css`
      background: transparent;
      border: 1px solid #e2e5ed;
      box-shadow: none;
      &:hover {
        background-image: none;
        border: none;
        box-shadow: none;
        border: 1px solid #cacedb;
      }
      &:focus {
        background-image: none;
        border: 1px solid #bdc2d1;
      }
    `}

  // Button without background and without hover or effect
  ${({ clear }) =>
    clear &&
    css`
      background: transparent;
      border: none;
      outline: none;
      box-shadow: none;
      color: ${({ theme }) => theme.buttonWrapper.clear.text};
      &:hover {
        color: ${({ theme }) => theme.buttonWrapper.clear.textHover};
      }
    `}

  &:disabled {
    cursor: not-allowed;
    &.icon {
      opacity: 0.3;
    }
    background: #747578;
    border: 1px solid #696A6E;
    box-shadow: inset 0 2px 0 0 rgba(255, 255, 255, 0.06);
    border-radius: 4px;
    color: $textSecondary;
  } //icon inisde the button

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

ButtonWrapper.defaultProps = {
  theme: {
    buttonWrapper: {
      primary: {
        background: `${Colours.PRIMARY}`
      },
      warning: {
        background: `${Colours.WARNING}`
      },
      success: {
        background: `${Colours.SUCCESS}`
      },
      clear: {
        text: `${Colours.TEXT_SECONDARY}`,
        textHover: `${Colours.TEXT_BASE}`
      }
    }
  }
};

export const ContentWrapper = styled.span`
  display: flex;
  flex-grow: 1;
  justify-content: center;
  align-items: center;
  height: 100%;

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;
