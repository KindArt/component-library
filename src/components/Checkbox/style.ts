import styled, { css } from 'styled-components';
import { Colours, Typography } from '../../constants/styles';

export const CheckboxIcon = styled.svg``;

export const Description = styled.p`
  margin-top: 2px;
  margin-left: 26px;
  color: ${Colours.TEXT_SECONDARY};
`;

interface ILabel {
  colour?: 'GREEN' | 'BLUE' | 'ORANGE' | 'RED';
}

export const Label = styled.div<ILabel>`
  color: ${Colours.TEXT_BASE};
  font-size: ${Typography.LABEL_FONT_SIZE};
  line-height: 19px;
  display: inline-flex;
  position: relative;
  cursor: pointer;
  &:hover {
    color: ${Colours.TEXT_MAIN};
  }
`;

export const CheckboxStyle = styled.input`
  width: 0px;
  height: 0px;
  overflow: hidden;
  position: absolute;
  visibility: hidden;

  &:checked + ${Label} {
    color: ${Colours.TEXT_MAIN};
    &:hover {
      color: ${Colours.TEXT_BASE};
    }
  }

  ${({ disabled }) =>
    disabled &&
    css`
      cursor: not-allowed;
      ${Label} {
        cursor: not-allowed;
      }
    `}
`;

interface IWrapper {
  toggle?: boolean;
  radio?: boolean;
  colour: 'GREEN' | 'BLUE' | 'ORANGE' | 'RED';
}

export const Wrapper = styled.label<IWrapper>`
  display: inline-block;
  line-height: 17px;
  user-select: none;

  ${({ toggle, radio, colour }) => {
    if (toggle) {
      return css`
        ${Description} {
          margin-left: 50px;
        }

        ${Label} {
          &:before,
          &:after {
            content: '';
            display: block;
            border-radius: 100px;
            transition: 0.2s;
          }
          &:before {
            width: 38px;
            height: 20px;
            background: ${Colours.BORDER_COLOUR};
            border: 1px solid ${Colours.BORDER_COLOUR};
            margin-right: 12px;
          }
          &:after {
            height: 16px;
            width: 16px;
            background: #ffffff;
            box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.15);
            position: absolute;
            top: 2px;
            left: 2px;
          }
        }

        ${CheckboxStyle}:checked+${Label} {
          &:before {
            border: 1px solid #2f9e3e;
            background-color: ${Colours[colour]};
            border-color: ${Colours[colour]};
          }
          &:after {
            transform: translateX(18px);
          }
        }
        ${CheckboxStyle}[disabled]+${Label} {
          &:after {
            opacity: 0.6;
          }
        }
      `;
    }

    if (radio) {
      return css`
        ${Label} {
          &:before,
          &:after {
            content: '';
            display: block;
            border-radius: 100px;
            transition: 0.2s;
          }
          &:before {
            margin-right: 10px;
            width: 16px;
            height: 16px;
            /* Radio: */
            background-image: linear-gradient(0deg, #f6f7f9 0%, #ffffff 100%);
            border: 1px solid #d8dce6;
            box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.05), inset 0 2px 0 0 rgba(255, 255, 255, 0.05);
          }
          &:after {
            height: 6px;
            width: 6px;
            background: #ffffff;
            position: absolute;
            top: 6px;
            left: 6px;
            opacity: 0;
          }
        }

        ${CheckboxStyle}:checked+${Label} {
          &:before {
            background-image: linear-gradient(0deg, ${Colours[colour]} 0%, ${Colours[colour]} 100%);
            border: 1px solid ${Colours[colour]};
            box-shadow: 0 1px 1px 0 rgba(19, 31, 21, 0.1), inset 0 2px 0 0 rgba(255, 255, 255, 0.06);
          }
          &:after {
            opacity: 1;
          }
        }
        ${CheckboxStyle}[disabled]+${Label} {
          &:before {
            background: rgba(#bcbcbc, 0.25);
          }
        }
      `;
    }

    if (!toggle && !radio) {
      return css`
        ${Label} {
          &:before {
            text-align: center;
            line-height: 14px;
            padding-left: 1px;
            color: #fff;
            width: 16px;
            height: 16px;
            content: '';
            font-size: 7px;
            background-color: #fff;
            border: 1px solid #dfe3e9;
            border-radius: 3px;
            margin-right: 10px;
            display: inline-block;
            transition: border 0.2s, background 0.2s;
            vertical-align: middle;
            transform: translateY(1px);
          }
          ${CheckboxIcon} {
            position: absolute;
            left: 5px;
            top: 7px;
            display: none;
          }
        }
        ${CheckboxStyle}[disabled] + ${Label}:before {
          /* Checkbox: */
          background-image: linear-gradient(0deg, #f6f7f9 0%, #ffffff 100%);
          border: 1px solid ${Colours.BORDER_COLOUR_DARKER};
          box-shadow: 0 1px 1px 0 rgba(22, 29, 37, 0.05), inset 0 2px 0 0 rgba(255, 255, 255, 0.05);
          border-radius: 3px;
          /* Rectangle: */
          background: rgba(#bcbcbc, 0.25);
        }
        ${CheckboxStyle}:checked + ${Label} {
          &:before {
            color: white;
            background-color: ${Colours[colour]};
            border-color: ${Colours[colour]};
          }
          ${CheckboxIcon} {
            display: block;
          }
        }
      `;
    }
  }}

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;
