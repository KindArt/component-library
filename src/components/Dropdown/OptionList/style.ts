import styled, { css } from 'styled-components';
import { Colours, Typography } from '../../../constants/styles';

export const Wrapper = styled.div`
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.08);
  border: 1px solid ${Colours.BORDER_COLOUR_DARKER};
  background: white;
  border-radius: 4px;
  transform: translateY(6px);
  max-height: 300px;
`;

export const List = styled.div`
  display: flex;
  flex-direction: column;
  max-height: 235px;
  position: relative;
  overflow-y: auto;
  -webkit-overflow-scrolling: touch;
`;

// needs the :focus
export const ButtonWrapperOverride = css`
  outline: none;
  cursor: pointer;
  background: transparent;
  border: none;
  border-radius: 0;
`;

export const ContentWrapperOverride = css`
  flex-grow: 0;
`;

export const OptionWrapper = styled.div`
  display: flex;
  align-items: center;
  & > * {
    min-width: 1px;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`;

export const Label = styled.div`
  font-size: ${Typography.DEFAULT_FONT_SIZE};
  pointer-events: none;
`;

export const Check = styled.div`
  color: ${Colours.GREEN};
  margin-left: 12px;
  flex-shrink: 0;
  position: relative;
  width: 5px;
  height: 5px;
  background: ${Colours.GREEN};
  border-radius: 50%;
`;
