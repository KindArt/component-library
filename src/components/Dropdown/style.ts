import styled, { css } from 'styled-components';
import { Colours, CssVariables } from '../../constants/styles';

// Dropdown Styles

export const Wrapper = styled.div`
  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

export const ValueRendered = css`
  width: 100%;
  text-align: left; // important for other selectors - :hover etc.
  position: relative;
  background: #ffffff;
  border: ${CssVariables.BORDER_DEFAULT};
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.08);
`;

export const ValueRenderedWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  & > * {
    flex-grow: 1;
    max-width: 100%;
    min-width: 1px;
  }
`;

export const ArrowWrapper = styled.div`
  flex-grow: 0;
  transition: 0.2s color;
  position: relative;
  width: 1px;
  height: 100%;
  margin-left: 12px;
  &:before,
  &:after {
    display: block;
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 0;
    height: 0;
    border-left: 3px solid transparent;
    border-right: 3px solid transparent;
    border-bottom: 3px solid ${Colours.TEXT_SECONDARY};
  }
  &:after {
    top: -4px;
  }
  &:before {
    top: 2px;
    transform: rotate(180deg);
  }
`;

const PlaceholderEmpty = css`
  color: ${Colours.TEXT_SECONDARY};
`;

interface IValueWrapper {
  placeholderEmpty?: boolean;
}

export const ValueWrapper = styled.div<IValueWrapper>`
  text-overflow: ellipsis;
  white-space: nowrap;
  ${({ placeholderEmpty }) => placeholderEmpty && PlaceholderEmpty}
`;

export const PlaceholderAndOther = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  & > *:first-child {
    min-width: 5px;
    text-overflow: ellipsis;
    white-space: nowrap;
  }
`;

export const PlusValue = styled.div`
  display: inline-block;
  background: ${Colours.BORDER_COLOUR_DARKER};
  border: ${CssVariables.BORDER_DEFAULT};
  height: 20px;
  border-radius: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 5px;
  min-width: 30px;
  margin-left: 10px;
  font-size: 0.7rem;
`;

export const ErrorMessage = styled.p`
  color: ${({ theme }) => theme.errorMessage.color};
  margin-top: 6px;
  line-height: 1.375rem;
`;

ErrorMessage.defaultProps = {
  theme: {
    errorMessage: {
      color: `${Colours.ERROR}`,
    },
  },
};
