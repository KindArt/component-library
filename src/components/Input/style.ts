import styled, { css } from 'styled-components';
import { HTMLProps } from 'react';
import { Colours, CssVariables } from '../../constants/styles';
import { IInputProps } from '.';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

export const InputWrapper = styled.div`
  display: flex;
  position: relative;
`;

interface IIconWrapper {
  iconPosition: 'left' | 'right';
}

export const IconWrapper = styled.span<IIconWrapper>`
  position: absolute;
  top: 50%;
  color: ${({ theme }) => theme.input.iconWrapperColour};
  transform: translateY(-50%);

  ${({ iconPosition }) =>
    iconPosition === 'left' &&
    css`
      left: 12px;
      & + ${InputField} {
        padding-left: 40px;
      }
    `}

  ${({ iconPosition }) =>
    iconPosition === 'right' &&
    css`
      // left: auto;
      right: 12px;
      & + ${InputField} {
        padding-right: 40px;
      }
    `}

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

IconWrapper.defaultProps = {
  theme: {
    input: {
      iconWrapperColour: Colours.BORDER_COLOUR_DARKER,
    },
  },
};

export const InputField = styled.input<IInputProps>`
  border: ${({ theme }) => theme.input.border};
  border-radius: 4px;
  flex: 1;
  padding: 10px 15px;
  font-size: 0.875rem;

  &:disabled {
    background: rgba(62, 63, 66, 0.3);
    cursor: not-allowed;
  }

  &:focus {
    border-color: ${({ theme }) => theme.input.focusColour};
  }

  ${({ errorMessage, theme }) =>
    errorMessage &&
    css`
      border-color: ${theme.input.errorBorderColor};
    `}

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

InputField.defaultProps = {
  theme: {
    input: {
      border: `${CssVariables.BORDER_DEFAULT}`,
      errorBorderColor: `${Colours.ERROR}`,
      focusColour: `${Colours.PRIMARY}`,
    },
  },
};

export const ErrorMessage = styled.p`
  color: ${({ theme }) => theme.errorMessage.color};
  margin-top: 6px;
  margin-bottom: 0;
  line-height: 1.375rem;

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

ErrorMessage.defaultProps = {
  theme: {
    errorMessage: {
      color: `${Colours.ERROR}`,
    },
  },
};
