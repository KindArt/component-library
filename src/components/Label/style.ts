import styled, { css } from 'styled-components';
import { Colours } from '../../constants/styles';

export const LabelWrapper = styled.span`
  font-weight: 500;
  font-size: 0.75rem;
  color: ${({ theme }) => theme.label.colour};
  line-height: 1.125rem;
  text-transform: uppercase;
  & + * {
    margin-top: 0.5rem;
  }

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

LabelWrapper.defaultProps = {
  theme: {
    label: {
      colour: `${Colours.TEXT_SECONDARY}`,
    },
  },
};
