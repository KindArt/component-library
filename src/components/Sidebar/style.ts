import styled, { css } from 'styled-components';
import { Colours } from '../../constants/styles';

interface ISidebarWrapperProps {
  height?: string;
  width?: string;
  isVisible: boolean;
}

export const SidebarWrapper = styled.div<ISidebarWrapperProps>`
  z-index: 99;
  display: flex;
  flex-direction: column;
  position: fixed;
  top: 0;
  left: 0;
  overflow-x: hidden;
  border-right: 1px solid ${Colours.BORDER_COLOUR};
  background-color: ${({ theme }) => theme.sidebar.backgroundColour};
  height: 100vh;
  width: 70vw;

  ${({ isVisible }) =>
    !isVisible &&
    css`
      display: none;
    `}

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

SidebarWrapper.defaultProps = {
  theme: {
    sidebar: {
      backgroundColour: '#252529',
    },
  },
};

export const CloseButton = styled.div`
  color: ${({ theme }) => theme.sidebar.cross.colour};
  position: absolute;
  top: 0;
  right: 25px;
  font-size: 36px;
`;

CloseButton.defaultProps = {
  theme: {
    sidebar: {
      cross: {
        colour: '#FFF',
      },
    },
  },
};

export const TopContent = styled.div``;

export const Content = styled.div`
  margin-top: 20px;
`;

export const Title = styled.h1`
  margin: 30px 0 0 0;
  text-transform: uppercase;
  font-size: 0.75rem;
  color: ${Colours.TEXT_SECONDARY};
  padding-left: 20px;

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

export const LinksContainer = styled.div`
  margin-top: 10px;

  ${({ className }) =>
    className &&
    css`
      ${className}
    `}
`;

interface ILinkProps {
  isActive: boolean;
}

export const Link = styled.div<ILinkProps>`
  color: ${({ theme }) => theme.sidebar.links.textColour};
  display: flex;
  align-items: center;
  height: 45px;
  padding-left: ${({ isActive }) => (isActive ? '20px' : '24px')};

  ${({ isActive, theme }) =>
    isActive &&
    `
      border-left: 4px solid ${theme.sidebar.links.borderColour};
      border-top: 1px solid rgba(22, 101, 216, 0.2);
      border-bottom: 1px solid rgba(22, 101, 216, 0.2);
      background-color: rgba(22, 101, 216, 0.08);
  `}

  :hover {
    cursor: pointer;
    color: ${Colours.PRIMARY};
  }
`;

Link.defaultProps = {
  theme: {
    sidebar: {
      links: {
        textColour: '#FFF',
        borderColour: Colours.PRIMARY,
      },
    },
  },
};
