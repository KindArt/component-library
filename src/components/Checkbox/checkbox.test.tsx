import React from 'react';
import { shallow } from 'enzyme';
import { Checkbox } from '../';
import { Description } from './style';
import Check from './check';

describe('<Checkbox />', () => {
  let wrapper: any;
  beforeEach(() => {
    wrapper = shallow(<Checkbox name="test" />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Visuals', () => {
    it('should display a label if one is passed', () => {
      wrapper.setProps({ label: 'This is a label' });
      const actual = wrapper.find('[data-test-id="Checkbox-label"]').text();
      const expected = 'This is a label';
      expect(actual).toEqual(expected);
    });

    it('should display a description if one is passed', () => {
      wrapper.setProps({ description: 'This is a description' });
      const actual = wrapper.find(Description).text();
      const expected = 'This is a description';
      expect(actual).toEqual(expected);
    });

    it('should display a check by default', () => {
      expect(wrapper.find(Check).length).toEqual(1);
    });

    it('should not display a check if toggle is passed', () => {
      wrapper.setProps({ toggle: true });
      expect(wrapper.find(Check).length).toEqual(0);
    });

    it('should not display a check if radio is passed', () => {
      wrapper.setProps({ radio: true });
      expect(wrapper.find(Check).length).toEqual(0);
    });
  });
});
