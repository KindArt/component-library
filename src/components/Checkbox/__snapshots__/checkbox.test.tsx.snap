// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render 1`] = `
<styled.label
  colour="GREEN"
>
  <styled.input
    name="test"
    type="checkbox"
  />
  <styled.div
    colour="GREEN"
  >
    <Check />
  </styled.div>
</styled.label>
`;
